src/openrct2/peep/peep.c

```diff
@@ -10419,10 +10420,10 @@ static sint32 sub_693C9E(rct_peep *peep)
 			continue;
 		if (top_z < mapElement->base_height)
 			continue;
+		if ((mapElement->flags & MAP_ELEMENT_FLAG_GHOST))
```

Comment:
No this looks fine to me. It should ignore the element completely just like it ignores it if it doesn't match PATH, TRACK OR ENTRANCE.